### PR TITLE
Add getMaxPageNumber to PageNumberOutOfRangeException

### PR DESCRIPTION
**Pull request by mynameisbogdan**
Hello,

When using `page_out_of_range: 'throwException'` you can't figure out which is the last page, so this PR adds method`getMaxPageNumber` to PageNumberOutOfRangeException, so you can redirect to last page if you'll like. 

~~Also trying the fix the deprecation message with `LegacyEventDispatcherProxy` on sf 5.1.~~

### CODE REVIEW

**Comment by mynameisbogdan** [comment]
Test failing seem to be related to mongodb.

File: src/Knp/Component/Pager/Paginator.php [hunk 1]
```diff
@@ -198,7 +201,7 @@ public function connect(string $eventName, $listener, int $priority = 0): void
      */
     protected function dispatch(string $eventName, Event\Event $event): void
     {
-        if (!\class_exists(LegacyEventDispatcherProxy::class)) {
+        if (\class_exists(BaseEvent::class)) {
```
Comment (garak) [inline src/Knp/Component/Pager/Paginator.php:204]: Please revert this change. You can propose it in a separate PR.

**Review by garak — CHANGES_REQUESTED** [review]

File: src/Knp/Component/Pager/Paginator.php (hunk 1 above)
Comment (mynameisbogdan) [reply src/Knp/Component/Pager/Paginator.php]: Hello. Okay, sounds better.

**Review by garak — COMMENTED** [review]
I see that a test is failing

**Comment by mynameisbogdan** [comment]
Yes, regarding that. I noticed after I created this PR that when using `throwException` with 0 results would also thrown an exception. Do you want to keep the current behaviour or act more like #271 only when count items is greater than 0?

**Comment by garak** [comment]
This case is different. With 0 results, the only valid page should be the first one.

**Comment by mynameisbogdan** [comment]
You're very much correct. 👍 

I changed that condition, plus added a new test.

**Review by garak — APPROVED** [review]
